### PR TITLE
Add a test for hierarchical watchers deleting a hierarchy

### PR DESCRIPTION
**Pull request by wolfs**
This is similar to the test on Linux, only for the hierarchical
watchers.

### CODE REVIEW

File: src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy [hunk 1]
```diff
@@ -617,16 +618,17 @@ class BasicFileEventFunctionsTest extends AbstractFileEventFunctionsTest {
             : Platform.current().windows
             ? [event(MODIFIED, removedFile), event(REMOVED, removedFile, false), event(REMOVED, watchedDir)]
             : [event(REMOVED, removedFile), event(REMOVED, watchedDir)]
-        removedDir.deleteDir()
+        def directoryDeleted = removedDir.deleteDir()
 
         then:
         expectedChanges.await()
+        directoryDeleted == canDeleteDirectory
```
Comment (lptr) [inline src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy]: Maybe call it `expectDirectoryToBeRemoved`?

File: src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy [hunk 2]
```diff
@@ -617,16 +618,17 @@ class BasicFileEventFunctionsTest extends AbstractFileEventFunctionsTest {
             : Platform.current().windows
             ? [event(MODIFIED, removedFile), event(REMOVED, removedFile, false), event(REMOVED, watchedDir)]
             : [event(REMOVED, removedFile), event(REMOVED, watchedDir)]
-        removedDir.deleteDir()
+        def directoryDeleted = removedDir.deleteDir()
```
Comment (lptr) [inline src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy]: Would be nice to standardize on "remove" as the term.

File: src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy [hunk 3]
```diff
@@ -156,6 +156,43 @@ class FileEventFunctionsStressTest extends AbstractFileEventFunctionsTest {
...
+            void reportError(Throwable ex) {
+                ex.printStackTrace()
+                uncaughtFailureOnThread << ex
+            }
+        }
+
+        when:
+        def watcher = startNewWatcher(callback)
+        watcher.startWatching(watchedDir)
+        Thread.sleep(500)
```
Comment (lptr) [inline src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy]: Can we do better than sleeping here? If we want all the directories to be deleted, then we should have a latch doing that. If we want to stop in the middle of the process, we should have a latch for that. WDYT? Can we maybe do better than sleeping in other cases maybe?

Comment (lptr) [reply src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy]: Even if we leave the sleep here, there should be a comment explaining what we are waiting for. Same for other places where we sleep in stress tests.

File: src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy [hunk 4]
```diff
@@ -156,6 +156,43 @@ class FileEventFunctionsStressTest extends AbstractFileEventFunctionsTest {
...
+        assert watchedDir.deleteDir()
+
+        then:
+        watcher.close()
```
Comment (lptr) [inline src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy]: This is a strange `then:` clause. Should this be `cleanup:` instead?

File: src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy (hunk 1 above)
Comment (wolfs) [reply src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy]: I am using `removed` now consistently.

File: src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy (hunk 2 above)
Comment (wolfs) [reply src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy]: Done.

File: src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy (hunk 3 above)
Comment (wolfs) [reply src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy]: This sleep was for making sure no changes from creating the directories come in before we start deleting them. I replaced it by `waitForChangeEventLatency()`.

File: src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy (hunk 4 above)
Comment (wolfs) [reply src/test/groovy/net/rubygrapefruit/platform/file/FileEventFunctionsStressTest.groovy]: I moved it to the `when` clause and now use `noExceptionThrown()` in the `then` clause.

**Comment by wolfs** [comment]
@lptr PTAL!

**Pull request merged**